ketcher-autotests/tests/Structure-Creating-&-Editing/SMARTS-attributes/Bond-attributes/bond-attributes.spec.ts

```diff
@@ -224,4 +242,181 @@ test.describe('Checking bond attributes in SMARTS format', () => {
       '[#6](-[#6])(-[#6])~,=[#6]',
     );
   });
+
+  test('Setting custom query - directional bond "up or unspecified"', async ({
+    page,
+  }) => {
+    /**
+     * Test case: https://github.com/epam/Indigo/issues/1281
+     * Description: "or unspecified"("?") quantifier should be supported
+     */
+    await setAndCheckBondProperties(
+      page,
+      setCustomQueryForBond,
+      '/?',
+      '[#6](-[#6])(-[#6])/?[#6]',
+    );
+  });
+});
+
+test.describe('Checking converting bond attributes to custom query', () => {
+  test.beforeEach(async ({ page }) => {
+    const numberOfBond = 2;
+    await waitForPageInit(page);
+    await drawStructure(page);
+    await page.keyboard.press('Escape');
+    await doubleClickOnBond(page, BondType.SINGLE, numberOfBond);
+    await waitForBondPropsModal(page);
+  });
+
+  test('Converting Topology = "Either" and Type = "Single" to custom query', async ({
+    page,
+  }) => {
+    /**
+     * Test case: https://github.com/epam/ketcher/issues/3372
+     * Description: If topology = "Either" then customQuery should be empty e.g. Type=Single, Topology=Either => customQuery=-
+     */
+    const expectedValue = '-';
+    await setBondType(page, 'Single-option');
+    await setBondTopology(page, 'Either-option');
+    await page.getByTestId('custom-query-checkbox').check();
+    const customQueryInput = page.getByTestId('bond-custom-query');
+    await expect(customQueryInput).toHaveValue(expectedValue);
+    await takeEditorScreenshot(page);
```

Comment:
These operations also are used more than once, maybe a common function could be extracted.